### ISSUE TITLE
Possibility to lock a task

### ISSUE THREAD

**Issue by dsimonet** [bardsoftware/ganttproject#1695]
If possible, add option to lock position (date) of a task. This way critical milestone couldn't be moved even if child of an another task.

For example, date of event or a show that can't be moved.

**Comment by dbarashev** [comment]
> couldn't be moved even if child of an another task.

Why don't you pull it to the top level then?
What shall happen when the whole project prior to milestone can't be completed without moving the milestone (because of task durations) ? 

Just in case if you are looking for backward scheduling, see issue #468

**Comment by dsimonet** [comment]
First, thanks for you answer. This is for milestone imposed to me by client or event outside of my project. If a make dependency from a task to this milestone, if my task take take longer time or if i move it, my milestone will move too! This behaviors is not always good. Because, some time I just didn't see this milestone follow and moved.

For example very important day like wedding can't be moved. So everything before have to fit before this date. And this date can't move...

**Comment by dbarashev** [comment]
>  if my task take take longer time or if i move it, my milestone will move too!

So what it is supposed to do if preceding tasks take longer time? It can't just stay where it was, because this way we miss the whole point of scheduling tasks.

**Comment by dsimonet** [comment]
Leave an error, becoming red ? excluding from critical path ? Yes it is strange behaviors. But if not, I have no way to create a point of event dissociated from task and not on the critical path. In fact I just want to create a mark in my project.
But If I create a milestone disconnected from my project I can't see critical path.

**Comment by dsimonet** [comment]
In fact there is only few type of project view... I know when my project ends and I want to put everything before and see when I have to start or if I have to much task ? Or the opposite, I want to start as soon as possible and know where it ends ? I kind of way, a lcoked task is a date where everything have to fit before... event if some task start before today... that just mean there is too much task or end date is too close.

**Comment by dbarashev** [comment]
The ultimate goal of planning in GP is to be able to say when the project completes.

 In case when you have a priori knowledge of the deadline date,

> So everything before have to fit before this date. And this date can't move...

 planning shall predict when the project should start to be able to meet the deadline. Maybe that's what you actually need in the context of planning a wedding? 

Or you don't want backward scheduling and just want to see a notification saying "hey this milestone was supposed to be at day X but now, whatever is the reason, it moved to day X+1, your action is  required!" ?

**Comment by dsimonet** [comment]
Yes but this popup gonna trow every modification... In fact my two needs are connected : 
- If a task is locked, every start-finish task needed gonna be before. and this task gonna be on the critical path.
- if as task is locked and don't want to be on the critical path. We have to exclude it from critical path. This way this task is fully independent. It just a day mark.
- other task work as usual around...

**Comment by dsimonet** [comment]
Anyway, this is just suggestion. Thanks for you help :-)
